### PR TITLE
feature/HW-52170 key and field gql integration

### PR DESCRIPTION
**Pull request by peter-joseph**
HW-52169 & HW-52170

### CODE REVIEW

File: ui/src/main/java/com/hyperwallet/android/ui/repository/TransferMethodConfigurationRepositoryImpl.java [hunk 1]
```diff
@@ -136,38 +137,53 @@ public Handler getHandler() {
...
+        mTransferMethodConfigurationKey = null;
     }
 
     @Override
     public void refreshFields() {
         mFieldMap.clear();
     }
+
+    @Nullable
+    private String getProcessingTime(String country, String currency, String transferMethodType) {
```
Comment (azakrevska-epam) [inline ui/src/main/java/com/hyperwallet/android/ui/repository/TransferMethodConfigurationRepositoryImpl.java]: @peter-joseph What about tests for getProcessingTime? Maybe we can remove private (access modifier)

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java [hunk 2]
```diff
@@ -289,26 +283,50 @@ public void notifyTransferMethodAdded(@NonNull final HyperwalletTransferMethod t
...
-                    mWidgetInputStateHashMap.put(widget.getName(), widget.getWidgetInputState());
+            Locale locale = new Locale.Builder().setRegion(mCountry).build();
+            // group
+            for (HyperwalletFieldGroup group : fields) {
+                String sectionHeaderText;
+                if (group.getGroupName().equals(HyperwalletFieldGroup.GroupTypes.ACCOUNT_INFORMATION)) {
+                    sectionHeaderText = requireContext().getResources()
+                            .getString(R.string.account_information_section_header, locale.getDisplayName(), mCurrency);
+                } else {
+                    sectionHeaderText = requireContext().getString(requireContext().getResources()
```
Comment (azakrevska-epam) [inline ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java]: @peter-joseph  there is we have a problem: what if we don't have such resource?
I propose at least:
 final int stringIdentifier = requireContext().getResources().getIdentifier(
                            group.getGroupName().toLowerCase(Locale.ROOT), "string", requireContext().getPackageName());
                    sectionHeaderText = stringIdentifier == 0 ? group.getGroupName() : requireContext().getString(
                            requireContext().getResources()
                                    .getIdentifier(group.getGroupName().toLowerCase(Locale.ROOT), "string",
                                            requireContext().getPackageName()));

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java [hunk 3]
```diff
@@ -289,26 +283,50 @@ public void notifyTransferMethodAdded(@NonNull final HyperwalletTransferMethod t
...
+                View sectionHeader = LayoutInflater.from(mDynamicContainer.getContext())
+                        .inflate(R.layout.item_widget_section_header, mDynamicContainer, false);
+                TextView sectionTitle = sectionHeader.findViewById(R.id.section_header_title);
+                sectionTitle.setText(sectionHeaderText);
+                sectionHeader.setId(View.generateViewId());
+                previousView = sectionHeader.getId();
+                mDynamicContainer.addView(sectionHeader);
+
+                // group fields
+                for (final HyperwalletField field : group.getFields()) {
```
Comment (azakrevska-epam) [inline ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java]: @peter-joseph  what should we do if there is no fields (they are nulablle) maybe we can add :
if (group.getFields() != null)

File: ui/src/main/java/com/hyperwallet/android/ui/repository/TransferMethodConfigurationRepositoryImpl.java (hunk 1 above)
Comment (peter-joseph) [reply ui/src/main/java/com/hyperwallet/android/ui/repository/TransferMethodConfigurationRepositoryImpl.java]: That's a temporary method, I added todo in other PR

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java (hunk 3 above)
Comment (peter-joseph) [reply ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java]: its a mis configuration

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java (hunk 2 above)
Comment (peter-joseph) [reply ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java]: no, I need parameters to passed to the other resource

**Pull request closed without merging**